src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -45,10 +48,32 @@ public AddCommand parse(String args) throws ParseException {
                         PREFIX_LICENCE_PLATE, PREFIX_ADDRESS, PREFIX_TAG, PREFIX_POLICY_NUMBER,
                         PREFIX_POLICY_ISSUE_DATE, PREFIX_POLICY_EXPIRY_DATE);
 
+        if (!argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_PREAMBLE_DETECTED));
+        }
+
         if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_NRIC,
-                PREFIX_LICENCE_PLATE, PREFIX_ADDRESS)
-                || !argMultimap.getPreamble().isEmpty()) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
+                PREFIX_LICENCE_PLATE, PREFIX_ADDRESS)) {
+            String errorMessage = MESSAGE_MISSING_FIELDS_FOR_ADD_COMMAND;
+            if (argMultimap.getValue(PREFIX_NAME).isEmpty()) {
+                errorMessage += "- Name(n/) ";
+            }
+            if (argMultimap.getValue(PREFIX_PHONE).isEmpty()) {
+                errorMessage += "- Phone(p/) ";
```

Comment:
Hence following Chai Yin suggestion, i will make the necessary changes here

```suggestion
                errorMessage += "- Phone(" + PREFIX_PHONE + ") ";
```